class-components/src/components/BottomSection/BottomSection.tsx

```diff
@@ -0,0 +1,145 @@
+import React, { useEffect, useState } from 'react';
+import { useLocation, useNavigate } from 'react-router-dom';
+import { SearchState, Item, ApiLink, ItemType } from './BottomSection.type';
+import {
+  isItemPeople,
+  formatPeople,
+  formatFilms,
+  formatPlanets,
+  formatSpecies,
+  formatStarship,
+  isItemFilms,
+  isItemPlanets,
+  isItemSpecies,
+  isItemStarship,
+} from './helpers';
+import Pagination from '../Pagination/Pagination';
+// import DetailSection from '../DetailSection/DetailSection';
+
+import './BottomSection.css';
+
+const ITEMS_PER_PAGE = 10;
+
+const useQuery = () => {
+  return new URLSearchParams(useLocation().search);
+};
```

Comment:
Why not use `useSearchParams `?
https://reactrouter.com/en/main/hooks/use-search-params#usesearchparams